### PR TITLE
doc: Improve documentation regarding OpenNebula integration with ceph

### PR DESCRIPTION
**Pull request by dann1**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
@ljflores: These docs changes look okay to me, but I don't think that I should be involved in changes to .gitignore without the consent of someone like you. The docs changes here mostly remove whitespace at the ends of lines, and occasionally mention OpenNebula in lists of other management utilities.

Let me know what you think.

**Comment by anthonyeleven** [comment]
Thanks for contributing!  LGTM -- @zdover23 ?

**Comment by zdover23** [comment]
> Thanks for contributing! LGTM -- @zdover23 ?

@anthonyeleven: I think that the docs changes are good, but I'm not going to mess with .gitignore without asking people on the leadership team. When the work week starts again, I expect that we'll get the thumbs-up and it'll be time to merge and backport this.

File: doc/install/index.rst [hunk 1]
```diff
@@ -59,6 +59,8 @@ tool that can be used to quickly deploy clusters. It is deprecated.
 
 `github.com/openstack/puppet-ceph <https://github.com/openstack/puppet-ceph>`_  installs Ceph via Puppet.
 
+`opennebula HCI clusters <https://docs.opennebula.io/stable/provision_clusters/hci_clusters/overview.html>`_ installs Ceph on different cloud providers
```
Comment (idryomov) [inline doc/install/index.rst:62]: A dot is missing at the end of the sentence.  Also, should opennebula be capitalized here?
```suggestion
`OpenNebula HCI Clusters <https://docs.opennebula.io/stable/provision_clusters/hci_clusters/overview.html>`_ installs Ceph on different cloud providers.
```

**Comment by idryomov** [comment]
> I think that the docs changes are good, but I'm not going to mess with .gitignore without asking people on the leadership team.

No objections for the `.gitignore` change.  Most folks are using a separate build directory which is ignored altogether and/or not building docs, so this went unnoticed.

**Comment by idryomov** [comment]
> so this went unnoticed.

It looks like @zdover23 actually ran into this but chalked it off as something related to submodules: https://github.com/ceph/ceph/commit/9382a90342f96cabab4b7caf816d9381f77e7f46.  This `.gitignore` change addresses:
```
            Untracked files:
              (use "git add <file>..." to include in what will be committed)
            src/pybind/cephfs/build/
            src/pybind/cephfs/cephfs.c
            src/pybind/cephfs/cephfs.egg-info/
            src/pybind/rados/build/
            src/pybind/rados/rados.c
            src/pybind/rados/rados.egg-info/
            src/pybind/rbd/build/
            src/pybind/rbd/rbd.c
            src/pybind/rbd/rbd.egg-info/
            src/pybind/rgw/build/
            src/pybind/rgw/rgw.c
            src/pybind/rgw/rgw.egg-info/
```
An outdated submodule would look differently.

File: doc/install/index.rst (hunk 1 above)
Comment (anthonyeleven) [reply doc/install/index.rst:62]: ... install Ceph on various cloud platforms.

**Comment by zdover23** [comment]
@idryomov:

> > so this went unnoticed.
> 
> It looks like @zdover23 actually ran into this but chalked it off as something related to submodules: [9382a90](https://github.com/ceph/ceph/commit/9382a90342f96cabab4b7caf816d9381f77e7f46). This `.gitignore` change addresses:
> 
> ```
>             Untracked files:
>               (use "git add <file>..." to include in what will be committed)
>             src/pybind/cephfs/build/
>             src/pybind/cephfs/cephfs.c
>             src/pybind/cephfs/cephfs.egg-info/
>             src/pybind/rados/build/
>             src/pybind/rados/rados.c
>             src/pybind/rados/rados.egg-info/
>             src/pybind/rbd/build/
>             src/pybind/rbd/rbd.c
>             src/pybind/rbd/rbd.egg-info/
>             src/pybind/rgw/build/
>             src/pybind/rgw/rgw.c
>             src/pybind/rgw/rgw.egg-info/
> ```
> 
> An outdated submodule would look differently.

It's true. A year ago, I didn't understand that this wasn't due to stale submodules. My understanding was based on antiquated and possibly even-in-the-first-instance misleading documentation and my own erroneous surmise.

I'll make all the changes here and clean this up and squash the changes and await the merge and then backport this. Later in the week, I'll remove the erroneous submodule-updating procedure.

Good looking out, Ilya.

File: doc/install/index.rst (hunk 1 above)
Comment (zdover23) [reply doc/install/index.rst:62]: Accepted.

**Comment by zdover23** [comment]
jenkins test api

**Review by zdover23 — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/54938 - Reef backport
https://github.com/ceph/ceph/pull/54939 - Quincy backport

**Comment by zdover23** [comment]
> > so this went unnoticed.
> 
> It looks like @zdover23 actually ran into this but chalked it off as something related to submodules: [9382a90](https://github.com/ceph/ceph/commit/9382a90342f96cabab4b7caf816d9381f77e7f46). This `.gitignore` change addresses:
> 
> ```
>             Untracked files:
>               (use "git add <file>..." to include in what will be committed)
>             src/pybind/cephfs/build/
>             src/pybind/cephfs/cephfs.c
>             src/pybind/cephfs/cephfs.egg-info/
>             src/pybind/rados/build/
>             src/pybind/rados/rados.c
>             src/pybind/rados/rados.egg-info/
>             src/pybind/rbd/build/
>             src/pybind/rbd/rbd.c
>             src/pybind/rbd/rbd.egg-info/
>             src/pybind/rgw/build/
>             src/pybind/rgw/rgw.c
>             src/pybind/rgw/rgw.egg-info/
> ```
> 
> An outdated submodule would look differently.

This is fixed here:

https://github.com/ceph/ceph/pull/54960

**Pull request merged**